src/runtime/TemporalObject.cpp

```diff
@@ -1318,9 +1365,58 @@ TemporalPlainYearMonth::TemporalPlainYearMonth(ExecutionState& state)
 {
 }
 
-TemporalPlainYearMonth::TemporalPlainYearMonth(ExecutionState& state, Object* proto)
+TemporalPlainYearMonth::TemporalPlainYearMonth(ExecutionState& state, Object* proto, int isoYear, int isoMonth, Object* calendar, int referenceISODay)
     : Temporal(state, proto)
+    , m_isoYear(isoYear)
+    , m_isoMonth(isoMonth)
+    , m_calendar(calendar)
+    , m_referenceISODay(referenceISODay)
+{
+}
+Value TemporalPlainYearMonth::createTemporalYearMonth(ExecutionState& state, int isoYear, int isoMonth, const Value& calendar, int referenceISODay, Optional<Object*> newTarget)
+{
+    ASSERT(calendar.isObject());
+
+    if (!TemporalPlainDate::isValidISODate(state, isoYear, isoMonth, referenceISODay)) {
+        ErrorObject::throwBuiltinError(state, ErrorObject::RangeError, "Invalid date");
+    }
+
+    if (!TemporalPlainYearMonth::isoYearMonthWithinLimits(isoYear, isoMonth)) {
+        ErrorObject::throwBuiltinError(state, ErrorObject::RangeError, "Invalid date");
+    }
+
+    if (!newTarget.hasValue()) {
+        newTarget = state.context()->globalObject()->temporal()->asTemporalObject()->getTemporalPlainYearMonthPrototype();
+    }
+
+    return new TemporalPlainYearMonth(state, newTarget.value(), isoYear, isoMonth, calendar.asObject(), referenceISODay);
+}
+
+bool TemporalPlainYearMonth::isoYearMonthWithinLimits(int isoYear, int isoMonth)
 {
+    return !(isoYear < -271821 || isoYear > 275760) || !(isoYear == -271821 && isoMonth < 4) || !(isoYear == 275760 && isoMonth > 9);
```

Comment:
Don't you need to validate `isoMonth` ?

Reply:
No because, any month is valid month if the years are within the range.

Reply:
The isoYear == -271822 will be true for `!(isoYear == -271821 && isoMonth < 4)`. Is this intentional?